lib/ansible/modules/cloud/amazon/kinesis_stream.py

```diff
@@ -804,6 +858,36 @@ def update(client, current_stream, stream_name, retention_period=None,
             )
             return success, changed, err_msg
 
+    if current_stream['OpenShardsCount'] != number_of_shards:
+        success, err_msg = (
+            update_shard_count(client, stream_name, number_of_shards, check_mode=check_mode)
+        )
+
+        if not success:
+            return success, changed, err_msg
+
+        changed = True
```

Comment:
On [line 820](https://github.com/ansible/ansible/pull/24805/files#diff-fc1fa6f27a9f3a0497153cf979c2732fL820) it's checking for not changed.